substrate/frame/support/src/traits/misc.rs

```diff
@@ -927,11 +929,65 @@ where
 	Signature: TypeInfo,
 	Extra: TypeInfo,
 {
-	fn call(&self) -> &Self::Call {
+	type Call = Call;
+
+	fn call(&self) -> &Call {
 		&self.function
 	}
 }
 
+pub trait InherentBuilder: ExtrinsicCall {
```

Comment:
Is this trait useful? it is only used once in the macro expansion, we can call `UnchechecedExtrinsic::new_bare()` there directly?

Reply:
The [`frame-support` proc macros use the generic `Block::Extrinsic`](https://github.com/paritytech/polkadot-sdk/blob/e6bd9205432bb524e94c9bd13048d645ec9aa5c7/substrate/frame/support/procedural/src/construct_runtime/mod.rs#L390), which isn't necessarily `UncheckedExtrinsic`. This trait is present to allow people building with substrate to use a different extrinsic type with the proc macros.